services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -994,7 +1068,46 @@ class ApplicationServiceITest : DatabaseTest() {
         assertThat(applications.map { it.applicationIdentifier })
             .containsExactlyInAnyOrder("JS2300082", "JS2300084")
     }
-    // }
+
+    @Test
+    fun `Creating an application without hankeTunnus fails`() {
+        assertThrows<HankeNotFoundException> {
+            applicationService.create(
+                AlluDataFactory.createApplication(id = null, hankeTunnus = null),
+                username
+            )
+        }
+    }
+
+    @Test
+    fun `Creating an application with incorrect hankeTunnus fails`() {
+        assertThrows<HankeNotFoundException> {
+            applicationService.create(
+                AlluDataFactory.createApplication(id = null, hankeTunnus = "1234"),
+                username
+            )
+        }
+    }
+
+    @Test
+    fun `hankeTunnus cannot be changed on update`() {
```

Comment:
Should there be an update call here? Or what is this testing?

Reply:
Update does not send hanke infromation again.

Reply:
I think I understand this test now after the changes. It checks that the entity relations are set up in a way that the hanke can't be changed for an application even if we try.

I think this test is in the wrong place, since it doesn't test any ApplicationService functionality.

I'm not sure how I feel about the change in hanke being just silently ignored. Usually, when we do something stupid, we want to know about it. If someone writes code that tries to change the hanke of an existing application, it should fail loudly so the author knows they made an oopsie.

Reply:
I removed the test.

Reply:
It's still there?

Reply:
Should not be.

Reply:
Isn't this the same test slightly updated?
```
    @Test
    fun `Hanke of an application cannot be changed`() {
        every { cableReportServiceAllu.create(any()) }.returns(2)

        val hanke = createHanke()
        val application =
            applicationService.create(
                AlluDataFactory.createApplication(id = null, hankeTunnus = hanke.hankeTunnus!!),
                username
            )

        assertThat(application.hankeTunnus).isEqualTo(hanke.hankeTunnus)

        val applicationEntity = applicationRepository.getOne(application.id!!)
        applicationEntity.hanke = hankeRepository.save(HankeEntity(hankeTunnus = "HAI-1111"))
        applicationRepository.save(applicationEntity)

        assertThat(applicationRepository.getOne(application.id!!).hanke.hankeTunnus)
            .isEqualTo(hanke.hankeTunnus)
    }
```